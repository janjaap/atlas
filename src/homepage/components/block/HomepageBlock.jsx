import React from 'react';
import PropTypes from 'prop-types';

import './_homepage-block.scss';

const HomepageBlock = (props) => (
  <div // eslint-disable-line jsx-a11y/no-static-element-interactions
    tabIndex="0"  // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
    className="homepage-block c-homepage__block c-homepage__block--tall"
    onKeyDown={(event) => { if (event.key === 'Enter') props.onBlockLinkClick(); }}
  >

    {props.children}

    {props.onBlockLinkClick && (
      <button
        tabIndex="-1"
        className={`c-homepage__block-button ${props.hasTallDescription ? 'c-homepage__block-button--tall' : ''}`}
        title={`Bekijk ${props.title}`}
        onClick={props.onBlockLinkClick}
      >
        <div className="o-btn--transparent">{props.title}</div>
        <div className="c-homepage__block-details">{props.description}</div>
      </button>
    )}
  </div>
);

HomepageBlock.propTypes = {
  onBlockLinkClick: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.object,
    PropTypes.node
  ]),
  description: PropTypes.string,
  hasTallDescription: PropTypes.bool,
  title: PropTypes.string
};

HomepageBlock.defaultProps = {
  onBlockLinkClick: undefined,
  children: '',
  description: '',
  hasTallDescription: false,
  title: ''
};

export default HomepageBlock;
