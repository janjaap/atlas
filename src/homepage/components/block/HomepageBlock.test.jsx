import React from 'react';
import { shallow } from 'enzyme';

import HomepageBlock from './HomepageBlock';

const mockFn = jest.fn();

describe('HomepageBlock', () => {
  it('should render with the default block-link', () => {
    const wrapper = shallow(
      <HomepageBlock
        onBlockLinkClick={mockFn}
        title={'test block'}
        description={'click here for the test link in the block'}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with the children and default block-link', () => {
    const wrapper = shallow(
      <HomepageBlock
        onBlockLinkClick={mockFn}
        title={'test block'}
        description={'click here for the test link in the block'}
      >
        <div>
          <span>This is a child of the homepageblock</span>
        </div>
      </HomepageBlock>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should navigate on enter ', () => {
    const wrapper = shallow(
      <HomepageBlock
        onBlockLinkClick={mockFn}
        title={'test block'}
        description={'click here for the test link in the block'}
      >
        <div>
          <span>This is a child of the homepageblock</span>
        </div>
      </HomepageBlock>
    );

    const mockEvent = { key: 'Enter' };
    wrapper.find('.homepage-block').simulate('keydown', mockEvent);
    expect(mockFn).toHaveBeenCalled();
  });
});
