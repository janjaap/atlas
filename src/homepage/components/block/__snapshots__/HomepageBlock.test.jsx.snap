// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageBlock should render with the children and default block-link 1`] = `
<div
  className="homepage-block c-homepage__block c-homepage__block--tall"
  onKeyDown={[Function]}
  tabIndex="0"
>
  <div>
    <span>
      This is a child of the homepageblock
    </span>
  </div>
  <button
    className="c-homepage__block-button "
    onClick={[MockFunction]}
    tabIndex="-1"
    title="Bekijk test block"
  >
    <div
      className="o-btn--transparent"
    >
      test block
    </div>
    <div
      className="c-homepage__block-details"
    >
      click here for the test link in the block
    </div>
  </button>
</div>
`;

exports[`HomepageBlock should render with the default block-link 1`] = `
<div
  className="homepage-block c-homepage__block c-homepage__block--tall"
  onKeyDown={[Function]}
  tabIndex="0"
>
  <button
    className="c-homepage__block-button "
    onClick={[MockFunction]}
    tabIndex="-1"
    title="Bekijk test block"
  >
    <div
      className="o-btn--transparent"
    >
      test block
    </div>
    <div
      className="c-homepage__block-details"
    >
      click here for the test link in the block
    </div>
  </button>
</div>
`;
